### PR TITLE
[SPARK-30532] DataFrameStatFunctions to work with TABLE.COLUMN syntax

### PR DESCRIPTION
**Pull request by kachayev**
### What changes were proposed in this pull request?
`DataFrameStatFunctions` now works correctly with fully qualified column name (Table.Column syntax) by properly resolving the name instead of relying on field names from schema, notably:
* `approxQuantile`
* `freqItems`
* `cov`
* `corr`

(other functions from `DataFrameStatFunctions` already work correctly).

See code examples below.

### Why are the changes needed?
With current implementation some stat functions are impossible to use when joining datasets with similar column names.

### Does this PR introduce any user-facing change?
Yes. Before the change, the following code would fail with `AnalysisException`.

```scala
scala> val df1 = sc.parallelize(0 to 10).toDF("num").as("table1")
df1: org.apache.spark.sql.Dataset[org.apache.spark.sql.Row] = [num: int]

scala> val df2 = sc.parallelize(0 to 10).toDF("num").as("table2")
df2: org.apache.spark.sql.Dataset[org.apache.spark.sql.Row] = [num: int]

scala> val dfx = df2.crossJoin(df1)
dfx: org.apache.spark.sql.DataFrame = [num: int, num: int]

scala> dfx.stat.approxQuantile("table1.num", Array(0.1), 0.0)
res0: Array[Double] = Array(1.0)

scala> dfx.stat.corr("table1.num", "table2.num")
res1: Double = 1.0

scala> dfx.stat.cov("table1.num", "table2.num")
res2: Double = 11.0

scala> dfx.stat.freqItems(Array("table1.num", "table2.num"))
res3: org.apache.spark.sql.DataFrame = [table1.num_freqItems: array<int>, table2.num_freqItems: array<int>]
```

### How was this patch tested?
Corresponding unit tests are added to `DataFrameStatSuite.scala` (marked as "SPARK-30532").

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by kachayev** [comment]
Fixed tests.

**Comment by kachayev** [comment]
Tests failed because of something that seems completely irrelevant (in `hive`). Digging deeper to understand why.

**Comment by HyukjinKwon** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala [hunk 1]
```diff
@@ -126,6 +126,19 @@ class DataFrameStatSuite extends QueryTest with SharedSparkSession {
     assert(math.abs(corr3 - 0.95723391394758572) < 1e-12)
   }
 
+  test("SPARK-30532 pearson correlation to understand fully-qualified column name") {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala]: can we merge these tests into one? Then we can share the below code
```
val df1 = spark.sparkContext.parallelize(0 to 10).toDF("num").as("table1")
val df2 = spark.sparkContext.parallelize(0 to 10).toDF("num").as("table2")
val dfx = df2.crossJoin(df1)
```

Comment (kachayev) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala]: Sure, I will submit update shortly.

Comment (kachayev) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala]: Done!

**Comment by kachayev** [comment]
Looks like tests failed for the same reason as previously. Could not replicate this locally (local build & tests are just fine).

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by cloud-fan** [comment]
retest  this please

**Comment by cloud-fan** [comment]
retest this please

**Comment by kachayev** [comment]
@cloud-fan Merged with latest master. Now tests work (previously they failed with unrelated issues).

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Pull request closed without merging**